test-programs/account-compression-test/tests/address_merkle_tree_tests.rs

```diff
@@ -391,7 +382,7 @@ async fn update_address_merkle_tree_failing_tests() {
         .unwrap();
     let changelog_index = address_merkle_tree.merkle_tree.changelog_index();
     // CHECK: 9 invalid changelog index
-    let invalid_changelog_index_low = changelog_index - 1;
+    let invalid_changelog_index_low = changelog_index - 2;
```

Comment:
why does changelog index -1 not work anymore?

Reply:
changelog - 1 still works because it is the changelog of the last Merkle tree update which is the basis for the proof that we are using.
An invalid changelog index can fail a transaction in two ways:
1. updates the proof with invalid values (for -1 it updates the proof with the same value, for -2 it inserts an invalid element)
2. encounters a leaf with the same index fails with CannotUpdateLeaf, because updating a proof with the changelog of the same index would overwrite the update during which the changelog was created.

Reply:
in other words its fine and it failed before because we encountered option 2.